src/main/java/io/cdap/plugin/dlp/DLPTransformPluginConfig.java

```diff
@@ -79,12 +86,32 @@
       .flatMap(Collection::stream)
       .collect(Collectors.toSet());
   }
+  /**
+   * Gets the custom template specified by user which can be either a templateId or a full custom template path
+   * @return String of templateId or custom template path
+   */
+  public String getCustomTemplate() {
+    return Strings.isNullOrEmpty(templateId) ? customTemplatePath :
+      String.format("projects/%s/inspectTemplates/%s", getProject(), templateId);
+  }
 
   public void validate(FailureCollector collector, Schema inputSchema) {
     if (customTemplateEnabled) {
-      if (!containsMacro("templateId") && templateId == null) {
-        collector.addFailure("Must specify template ID in order to use custom template", "")
-          .withConfigProperty("templateId");
+      if (!containsMacro(CUSTOM_TEMPLATE_PATH_NAME) && Strings.isNullOrEmpty(CUSTOM_TEMPLATE_PATH_NAME)) {
+        if (!containsMacro(TEMPLATE_ID_NAME) && Strings.isNullOrEmpty(TEMPLATE_ID_NAME)) {
+          collector.addFailure("Must specify template ID in order to use custom template", "")
+            .withConfigProperty(TEMPLATE_ID_NAME);
+        }
+      }
+      if (!containsMacro(TEMPLATE_ID_NAME) && Strings.isNullOrEmpty(TEMPLATE_ID_NAME)) {
```

Comment:
Is this condition same as above condition?

Reply:
No. This condition is the opposite of above.

Reply:
This condition is the opposite of above. It checks whether both of custom template fields are filled (not empty). I simplified the conditions in the last updated PR.